src/main/java/com/paneedah/weaponlib/WeaponSpawnEntity.java

```diff
@@ -82,12 +88,12 @@ protected void onImpact(final RayTraceResult position) {
             Explosion.createServerSideExplosion(weapon.getModContext(), world, this.getThrower(), this, position.hitVec.x, position.hitVec.y, position.hitVec.z, explosionRadius, false, true, isDestroyingBlocks, explosionParticleAgeCoefficient, smokeParticleAgeCoefficient, explosionParticleScaleCoefficient, smokeParticleScaleCoefficient, weapon.getModContext().getRegisteredTexture(explosionParticleTextureId), weapon.getModContext().getRegisteredTexture(smokeParticleTextureId), weapon.getModContext().getExplosionSound());
         } else if (position.entityHit != null) {
             if (this.getThrower() != null)
-                position.entityHit.attackEntityFrom(DamageSource.causeThrownDamage(this, this.getThrower()), damage);
+                position.entityHit.attackEntityFrom(new ProjectileDamageSource(this, this.getThrower()), damage);
             else
-                position.entityHit.attackEntityFrom(new DamageSource("thrown"), damage);
+                position.entityHit.attackEntityFrom(new DamageSource("arrow"), damage);
```

Comment:
We already said we won't be using 'arrow' as the damage source. It's not supposed to be like this.

Reply:
The source of damage is not an arrow, it's just that the type of death message is an arrow